### PR TITLE
Feature/gradient clipping v2

### PR DESCRIPTION
**Pull request by pldelisle**

### CODE REVIEW

File: kerosene/training/gradient_clipping_mixin.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+class GradientClippingMixin(object):
+
+    def __init__(self):
+        self._clipping_func = {
+            "norm": torch.nn.utils.clip_grad_norm_,
+            "value": torch.nn.utils.clip_grad_value_
+        }
+
+    def clip_gradients(self, parameters, clipping_func, params):
```
Comment (banctilrobitaille) [inline kerosene/training/gradient_clipping_mixin.py]: Remove method args

File: kerosene/training/trainers.py [hunk 2]
```diff
@@ -134,6 +138,7 @@ def train(self, mode=True):
 
     def step(self):
         if self._status is Status.TRAIN:
+            self.clip_gradients(self._model.parameters(), self._gradient_clipping_func, self._gradient_clipping_params)
```
Comment (banctilrobitaille) [inline kerosene/training/trainers.py]: Add a should_clip condition

File: kerosene/nn/utils/gradients.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    def __init__(self, model_parameters: Union[Iterable[torch.Tensor], torch.Tensor], max_norm: float,
+                 norm_type: Union[int, float]):
+        """"
+        Gradient norm clipping strategy.
+
+        Args:
+            max_norm (float or int): max norm of the gradients
+            norm_type (float or int): type of the used p-norm. Can be ``'inf'`` for infinity norm.
+        """
+        self._model_parameters = model_parameters
```
Comment (banctilrobitaille) [inline kerosene/nn/utils/gradients.py]: Should put it as a arg to the clip func. A strategy does not need parameters to exists :small_red_triangle:
